### PR TITLE
Check for report permission before rendering REPORTS menu item

### PR DESCRIPTION
**Pull request by ZacharyKlein**

### CODE REVIEW

File: web-ui/src/context/selectors.js [hunk 1]
```diff
@@ -40,6 +40,12 @@ export const selectIsAdmin = createSelector(
     userProfile && userProfile.role && userProfile.role.includes("ADMIN")
 );
 
+export const selectHasReportPermission = createSelector(
+    selectUserProfile,
+    (userProfile) =>
+        userProfile && userProfile.role && userProfile.permissions.any((permission) => permission.includes("REPORT"))
+);
+
```
Comment (mkimberlin) [inline web-ui/src/context/selectors.js:48]: @ZacharyKlein permissions for the current user are already loaded in state?

**Comment by mkimberlin** [comment]
@ZacharyKlein I would probably add a test or two for that new selector.

File: web-ui/src/context/selectors.js (hunk 1 above)
Comment (ZacharyKlein) [reply web-ui/src/context/selectors.js:48]: @mkimberlin I believe so, when I inspect the network response from `/member-profiles/current` contains this block:

```
"permissions": [
        {
            "id": "008f6641-0b0a-4e89-84f0-c580f912b80d",
            "permission": "CAN_VIEW_FEEDBACK_REQUEST"
        },
        {
            "id": "056e32b2-8d07-11ee-b9d1-0242ac120002",
            "permission": "CAN_VIEW_ANNIVERSARY_REPORT"
        },
```

**Comment by ZacharyKlein** [comment]
> @ZacharyKlein I would probably add a test or two for that new selector.

I verified that (with my latest changes) the snapshot test for the admin menu fails without a "_REPORT" permission, and passes when the permission is present. I also added a test for the new selectHasReportPermission selector (and fixed a bug in the process)

File: web-ui/src/components/menu/Menu.jsx [hunk 2]
```diff
@@ -137,7 +137,11 @@ function Menu() {
   const csrf = selectCsrfToken(state);
   const { id, workEmail } =
     userProfile && userProfile.memberProfile ? userProfile.memberProfile : {};
+
+  console.log("userProfile", userProfile);
+
```
Comment (mkimberlin) [inline web-ui/src/components/menu/Menu.jsx:142]: 😱

**Review by mkimberlin — APPROVED** [review]

**Pull request merged**